packages/forms/src/UIForm/utils/validation.js

```diff
@@ -19,28 +19,131 @@ export function validateValue(schema, value, properties, customValidationFn) {
 }
 
 /**
- * Validate values.
- * @param mergedSchema The merged schema array.
- * @param properties The values.
+ * Validate an array.
+ * @param mergedSchema The array schema
+ * @param value The value
+ * @param properties All the values
+ * @param customValidationFn A custom validation function
+ * that is applied on schema.customValidation = true
+ * @param deepValidation Validate the array values if set to true
+ * @returns {object} The validation result.
+ */
+export function validateArray(mergedSchema, value, properties, customValidationFn, deepValidation) {
+	const results = {};
+	const { key, items } = mergedSchema;
+
+	// validate array definition, not its sub-items here
+	const schemaWithoutItems = {
+		...mergedSchema,
+		schema: {
+			...mergedSchema.schema,
+			items: [],
+		},
+	};
+	const error = validateValue(schemaWithoutItems, value, properties, customValidationFn);
+	if (error) {
+		results[key] = error;
+	}
+
+	// validate each value of the array
+	if (deepValidation && value) {
+		for (let valueIndex = 0; valueIndex < value.length; valueIndex += 1) {
+			// adapt items schema with value index
+			const indexedItems = items.map((item) => {
+				const indexedKey = [...item.key];
```

Comment:
No I can't. the purpose of this code is to create the new schema key for the item. The new key contains the index, so it must be in the loop.